src/vm/costs/mod.rs

```diff
@@ -716,10 +753,16 @@ fn compute_cost(
             &cost_function_reference
         )))?;
 
-    let program = vec![
-        SymbolicExpression::atom(cost_function_reference.function_name[..].into()),
-        SymbolicExpression::atom_value(Value::UInt(input_size.into())),
-    ];
+    let mut program = vec![SymbolicExpression::atom(
+        cost_function_reference.function_name[..].into(),
+    )];
+
+    let mut args = input_sizes
+        .into_iter()
+        .map(|size| SymbolicExpression::atom_value(Value::UInt(size.into())))
+        .collect::<Vec<SymbolicExpression>>();
+
+    program.append(&mut args);
```

Comment:
Instead of using `into_iter/map/collect/append`, you can do the following:

```suggestion
    for input_size in input_sizes.iter() {
        program.append(SymbolicExpression::atom_value(Value::UInt(input_size as u128)));
    }
```

And then all these functions can accept `input_sizes: &[u64]`, which means you wouldn't need to call `.to_vec()` at line 860 (and other places can just do `&[arg]` instead of constructing a vec.